tests/generic/check_seed.sql

```diff
@@ -0,0 +1,79 @@
+-- this tests checks a model for every row in a seed file.
+-- you need to specify the matching columns and the columns to check for equality.
+
+{% test check_seed(model, seed_file, match_columns=[], check_columns=[]) %}
+    {# --jinja comment
+--    I wish we could do something dynamic like this below, but it is prohibited by having a local database connection.. :(
+--    {%- set unique_columns = config.get('unique_key') -%}
+--    {%- set seed_check_columns = dbt_utils.get_filtered_columns_in_relation(from=seed_file, except=unique_columns) -%}
+--    {%- set seed_matching_columns = dbt_utils.get_filtered_columns_in_relation(from=seed_file, except=seed_check_columns) -%}
+    --jinja comment #}
+    {{ config(severity = 'warn') }}
+    {%- set seed_check_columns = check_columns -%}
+    {%- set seed_matching_columns = match_columns -%}
+
+    with matched_records as (
+        select
+        {%- for column_name in seed_matching_columns %}
+        seed.{{column_name}} as seed_{{column_name}},
+        model.{{column_name}} as model_{{column_name}},
+        {% endfor -%}
+        {%- for column_name in seed_check_columns %}
+        seed.{{column_name}} as seed_{{column_name}},
+        model.{{column_name}} as model_{{column_name}} {% if not loop.last %},{% endif %}
+        {% endfor -%}
+        from {{seed_file}} seed
+        left join {{model}} model
+        ON 1=1
+        {%- for column_name in seed_matching_columns %}
+        AND seed.{{column_name}} = model.{{column_name}}
+        {% endfor -%}
+    ),
```

Comment:
I tested the performance on different join logic of seeds and the model:

1.  [seeds left join the model](https://dune.com/queries/1508699)
2. [seeds inner join the model](https://dune.com/queries/1508752)
3. [seeds left join (unique keys from seeds inner join the model)](https://dune.com/queries/1508732)

Seeds represent the smaller dataset, while models(in this case ethereum.transactions) represent the larger dataset/table.
If we use the [seeds inner join the model] approach, we can add another counter test at the same time.